externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -141,6 +156,42 @@ case class SparkSQLEngine(spark: SparkSession) extends Serverable("SparkSQLEngin
     }
   }
 
+  private[kyuubi] def startMetricsReporter(): Unit = {
+    val interval = conf.get(ENGINE_REPORT_INTERVAL)
+    val engineSpace = conf.get(HA_NAMESPACE)
+    val statusTracker = spark.sparkContext.statusTracker
+    val metricsSpace = s"/metrics$engineSpace"
+    val report: Runnable = () => {
+      if (!shutdown.get) {
+        val openSessionCount = backendService.sessionManager.getOpenSessionCount
+        val activeTask = statusTracker.getActiveStageIds()
+          .flatMap { stage =>
+            statusTracker.getStageInfo(stage).map(_.numActiveTasks)
+          }.sum
+        val engineMetrics = Map(
+          "openSessionCount" -> openSessionCount,
+          "activeTask" -> activeTask,
+          "poolId" -> engineSpace.split("-").last)
+        info(s"Spark engine has $openSessionCount open sessions and $activeTask active tasks.")
+        DiscoveryClientProvider.withDiscoveryClient(conf) { client =>
+          if (client.pathNonExists(metricsSpace)) {
+            client.create(metricsSpace, "PERSISTENT")
+          }
+          client.setData(
+            s"/metrics$engineSpace",
```

Comment:
is it feasible to append the "load" value in engine's info zonde, and update in place?
@yaooqinn how do you think of this path to store engine metrics?